acceptance/suite_test.go

```diff
@@ -0,0 +1,284 @@
+	}, "healthz", "8080/tcp")
+}
+
+func (s *Suite) waitForReady(template string, args ...interface{}) {
+	s.Require().NoError(
+		s.pool.Retry(
+			healthCheck(template, args...),
+		),
+	)
+}
+
+// testFilePath returns the path of this test file, without the trailing slash
+func (s *Suite) testFilePath() string {
+	_, testFilename, _, _ := runtime.Caller(0)
+	path := testFilename[:strings.LastIndex(testFilename, "/")]
+	return path
+}
+
+// healthCheck creates a health check function that will succeed once the url returns a 200 OK status
+// the template provided will be formatted-f with the args
+func healthCheck(template string, args ...interface{}) func() error {
+	url := fmt.Sprintf(template, args...)
+	return func() error {
+		resp, err := http.Get(url)
+		if err != nil {
+			return err
+		}
+		if resp.StatusCode != http.StatusOK {
+			return fmt.Errorf("status code isn't 200 OK, is %s", resp.Status)
+		}
+		return nil
+	}
+}
+
+// waitUntilElapsedAfterSuiteSetup waits until duration is elapsed after suite setup
+// this allows dynamic wait periods depending on the amount of tests running,
+// avoiding double waits when more than one test is running which in the end makes waiting period more deterministic.
+func (s *Suite) waitUntilElapsedAfterSuiteSetup(duration time.Duration) {
+	wait := time.Until(s.suiteReady.Add(duration))
+	s.T().Logf("Waiting %s", wait)
+	time.Sleep(wait)
+	s.T().Log("Done waiting")
+}
+
+// startContainer will run a container with the given options, run a health check until the resource is available,
+// and then return.
+func (s *Suite) startContainer(runOptions *dockertest.RunOptions, healthEndpoint, healthPort string) *dockertest.Resource {
+	// Make sure that there are no containers running from previous execution first
+	s.Require().NoError(s.pool.RemoveContainerByName(runOptions.Name))
+
+	image := fmt.Sprintf("%s:%s", runOptions.Repository, runOptions.Tag)
+	s.T().Logf("Starting %s from %s", runOptions.Name, image)
+
+	if runOptions.Tag == "local" {
+		_, err := s.pool.Client.InspectImage(image)
+		s.Require().NoError(err, "Could not find %s, have you run `make build-local`?", image)
+	}
+
+	resource, err := s.pool.RunWithOptions(runOptions)
+	s.Require().NoError(err)
+	fmt.Println("starting wait for ready")
+	s.waitForReady("http://%s:%s/"+healthEndpoint, s.cfg.Docker.Host, resource.GetPort(healthPort))
+	fmt.Println("done with wait for ready")
+	return resource
+}
+
+type httpClient struct {
+	endpoint    string
+	http_client *http.Client
+}
+
+func (pc httpClient) query(path string, orgID string) (statusCode int, respBody []byte, err error) {
+	req, err := http.NewRequest("GET", pc.endpoint+path, nil)
+	fmt.Println("path: ", path)
+	req.Header.Set("X-Scope-OrgID", orgID)
+	if err != nil {
+		return 0, nil, err
+	}
+	resp, err := pc.http_client.Do(req)
+	fmt.Println("Do response: ", resp)
+	if err != nil {
+		return 0, nil, err
+	}
+	defer resp.Body.Close()
+
+	respBody, _ = io.ReadAll(resp.Body)
```

Comment:
I think it was done like this in the reference code, but we should add an error check here instead of ignoring it

Reply:
Agree :+1: